requirements.txt

```diff
@@ -17,7 +18,10 @@ pylint==2.6.0
 python-dateutil==2.8.1
 pytz==2020.4
 regex==2020.11.13
+scipy==1.5.4
 six==1.15.0
+sklearn==0.0
+sparse_dot_topn==0.2.9
```

Comment:
My computer is really choking installing this. I'll troubleshoot it if you want me to actually use the code, but as of now, I'm reviewing for design, not performance

Reply:
Don't worry about it at all. If I have a problem I'll open an issue. I have spent three days on optimization, because I need more data to start running a good ML model. Tried looking at threading but I couldn't get the idea straight in my head and made the code actually take longer to run so that's out. I would be interested in your take on [Numba](https://numba.readthedocs.io/en/stable/) if you have ever messed with anything like that. May come in handy for the cleaning aspect.